src/Fmt_ast.ml

```diff
@@ -303,18 +303,41 @@ let rec fmt_longident (li: Longident.t) =
       cbox 0 (fmt_longident li1 $ wrap "(" ")" (fmt_longident li2))
 
 
-let fmt_constant const =
+let fmt_constant (conf: Conf.t) const =
+  let ocaml_whitespace = function
+    | '\n' | '\t' | '\r' | ' ' -> true
+    | _ -> false
+  in
+  let escape_char c =
+    let code = Char.to_int c in
+    match conf.Conf.escape_chars with
+    | _ when ocaml_whitespace c || Char.equal c '\\' -> str (Char.escaped c)
+    | _ when 0x20 <= code && code <= 0x7E -> char c
```

Comment:
For readability, how about defining a named function for this test. Something like `is_printing c` or `not (is_control_code c)`.

Reply:
And when a char is part of a string, then the `"` char needs to be escaped.